src/material/datepicker/datepicker-input.ts

```diff
@@ -123,25 +129,33 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
 
   /** The value of the input. */
   @Input()
-  get value(): D | null { return this._value; }
+  get value(): D | null { return this._value ? this._value.asDate() : null; }
   set value(value: D | null) {
-    value = this._dateAdapter.deserialize(value);
-    this._lastValueValid = !value || this._dateAdapter.isValid(value);
-    value = this._getValidDateOrNull(value);
-    const oldDate = this.value;
-    this._value = value;
-    this._formatValue(value);
-
+    value  = this._dateAdapter.deserialize(value);
+    const oldDate = this._value.asDate();
+    let isDifferent = false;
     if (!this._dateAdapter.sameDate(oldDate, value)) {
-      this._valueChange.emit(value);
+      isDifferent = true;
+    }
+    this._value.add(value);
```

Comment:
Can you add a `setDate` method to `MatSingleDateSelectionModel` and call there here instead. It's more semantically correct, since we're overwriting the selected value, not adding to the current selection.

Reply:
We changed add to setDate on the abstract class, and updated the classes accordingly. Please let us know if it's ok.

Reply:
Ah this isn't quite what I was asking for. That method _should_ be called `add`, its just that there should be a second method, only on `MatSingleDateSelectionModel` called `setDate`. The `MatDateRangeSelectionModel` will have a method called `setRange` instead. Even though `setDate` and `add` will do the same thing, I'd like to separate them out now in case they wind up needing subtly different behavior

Reply:
Got it. We opted for this change because we noticed `add` and `setDate` would be doing the same thing but it makes sense if we might change the behavior later.